### ISSUE TITLE
Build failure: python311Packages.onnx

### ISSUE THREAD

**Issue by nonetrix** [NixOS/nixpkgs#281065]
### Steps To Reproduce

1. build python3.11-onnx-1.15.0.drv
2. See failure

### Build log
https://gist.github.com/nonetrix/a7911ecc9fdfc19a814608dd49f1418d

### Additional context

Happened while updating with `nix flake update` and running `nixos-rebuild switch --flake .` when trying to build `livecaptions-0.4.1.drv`

### Notify maintainers

@wegank @happysalada

### Metadata

```console
nixdots on  master [!]
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.71, NixOS, 24.05 (Uakari), 24.05.20240108.317484b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"home-manager, nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
 ```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by nonetrix** [comment]
It seems the log file is incomplete, how do I get a full log file outside of the pager?

**Comment by happysalada** [comment]
this looks like an abseil-cpp error, this compilation seems to need cpp 17 and lacking it.
it seems the fix is going to be to have something like
```
    abseil-cpp = abseil-cpp.override {
      cxxStandard = "17";
    };
```
most likely in the argument passed to onnx.

**Comment by happysalada** [comment]
someone seems to be working on a fix though https://github.com/NixOS/nixpkgs/pull/281011

**Comment by cbourjau** [comment]
This seems related: https://github.com/onnx/onnx/issues/5563

Please feel free to suggest an edit in https://github.com/NixOS/nixpkgs/pull/281011 if you know how to address this issue!

**Comment by Scrumplex** [comment]
This was fixed by #281011
